spec/expression.dd

```diff
@@ -261,6 +261,18 @@ $(GNAME AssignExpression):
     occurs. The resulting expression is a modifiable lvalue.
     )
 
+	$(P Operands of an assign expression are evaluated before the assign expression. 
+	The expression)
+        --------------
+        a op= b
+        --------------
+
+        is exactly the same as:
+
+        --------------
+        a = a op b
```

Comment:
what if `a` has side-effects?

Reply:
Isn't that the point? Whether or not `a` has side effects, it would be evaluated once:

```
exp1 op= exp2;
```

=>

```
{
    auto tmp1 = exp1;
    auto tmp2 = exp2;
    tmp1 = tmp1 op tmp2;
}

That is, unless `exp1` resolves to a type with custom `opAssign`, or `opOpAssign`.

Reply:
Then in most cases, a can't be an lvalue in an op= expression, unless the result of a is a stable reference (iirc, `*a op= b`)